### ISSUE TITLE
Alias pattern in GraphQL::Pro::RelationConnection::Selects is too restrictive

### ISSUE THREAD

**Issue by bdurand** [rmosolgo/graphql-ruby#3558]
**Describe the bug**

The regular expression in `GraphQL::Pro::RelationConnection::Selects::ALIASED_SELECT` is too restrictive and only works with aliased columns. If you pass in a relation that has a select value using a derived value, the alias name will end up being aliased twice which is not valid SQL.

**Versions**

`graphql-pro` version: 1.18.0

**Steps to reproduce**

A sample select clause where this fails is something like this where you want to sort on a derived column:

```sql
SELECT things.*,
  CASE WHEN promoted THEN 1 ELSE 0 END AS promoted_order
FROM things
ORDER BY promoted_order DESC, created_at DESC
```

However, when running the pattern against that statement:

```ruby
GraphQL::Pro::RelationConnection::Selects::ALIASED_SELECT.match("CASE WHEN promoted THEN 1 ELSE 0 END AS promoted_order")
```

You get `nil` instead of the expected

```
#<MatchData
 "CASE WHEN promoted THEN 1 ELSE 0 END AS promoted_order"
 selection:"CASE WHEN promoted THEN 1 ELSE 0 END AS promoted_order"
 alias:"promoted_order">
```

This results in a SQL statement like this once the cursor aliases are added which causes an error because you cannot alias an alias in Postgrea:

```sql
SELECT things.*,
  CASE WHEN promoted THEN 1 ELSE 0 END AS promoted_order
  promoted_order AS cursor_0
  created_at AS cursor_1
FROM things
ORDER BY promoted_order DESC, created_at DESC
```

In our application, we are overriding the pattern to be less restrictive and allow derived values with

```
/\s*(?<selection>.*)\s+AS\s*(["`]?(?<alias>\w*)["`]?)/mi
```

**Comment by rmosolgo** [comment]
Thanks for the detailed write-up of the bug and the details about how you worked around it. I'll fix up graphql-pro to address this case and follow up here soon!

**Comment by rmosolgo** [comment]
Thanks again for reporting this. I just shipped a fix in graphql-pro v1.18.1. If you run into any more trouble, please let me know!
